File: OdontoFlowAPI/OdontFlow/OdontFlow.Domain/ViewModel/PriceList/PriceListViewModel.cs

using OdontFlow.Domain.Entities;

namespace OdontFlow.Domain.ViewModel.PriceList;

public class PriceListViewModel
{
    public Guid Id { get; set; }
    public string Name { get; set; } = default!;
    public string Category { get; set; } = default!; 
    public float Discount { get; set; }
    public int CategoryId { get; set; } = default!;
    public ICollection<PriceListItemViewModel> Items { get; set; } = new List<PriceListItemViewModel>();
    public bool Active { get; set; } = default!;
}

File: OdontoFlowWeb/OdontFlow/src/app/core/model/pricelist-view.model.ts
export interface PriceListViewModel {
    id?: string;                       
    name?: string; 
    category?: string;
    categoryId?: Number;
    discount?: number;
    active?:boolean;
  }
  
  export interface CreatePriceListDTO {
    name?: string;
    contact?: string;
    category?: number;
    categoryId?: Number;
    discount?: Number;   
    active?:boolean; 
  }
  
  export interface UpdatePriceListDTO extends CreatePriceListDTO {
    id: string | undefined;  
  }
  
